scalabel/label/from_mmpkl.py

```diff
@@ -0,0 +1,645 @@
+            location=(
+                float(label['kitti_format_pred']['location'][0][0]),
+                float(label['kitti_format_pred']['location'][0][1]) - y_cen_adjust,
+                float(label['kitti_format_pred']['location'][0][2]),
+            ),
+            dimension=(
+                float(label['kitti_format_pred']['dimensions'][0][0]),
+                float(label['kitti_format_pred']['dimensions'][0][1]),
+                float(label['kitti_format_pred']['dimensions'][0][2]),
+            ),
+            alpha=float(label['kitti_format_pred']['alpha'][0]),
+        )
+        
+        labels_dict[seq_id].append(
+            Label(
+                category=class_name,
+                box2d=box2d,
+                box3d=box3d,
+                id=str(track_id),
+                poly2d=None,
+                graph=None,
+                rle=None,
+            )
+        )
+
+    return labels_dict, trackid_maps, global_track_id
+
+
+def generate_labels_cam(labels: List[Label], offset: float) -> List[Label]:
+    """Adjust label with different cameras."""
+    labels_cam = []
+    for label in labels:
+        label_cam = copy.deepcopy(label)
+        assert label_cam.box3d is not None
+        label_cam.box3d.location = (
+            label_cam.box3d.location[0] + offset,
+            label_cam.box3d.location[1],
+            label_cam.box3d.location[2],
+        )
+        labels_cam.append(label_cam)
+
+    return labels_cam
+
+
+def find_nearest_lidar_frame(
+    velodyne_name: str,
+    velodyne_dir: str,
+    video_name: str,
+    frame_idx: int,
+    velodyne_names: List[str],
+) -> str:
+    """Find the nearest lidar frame to handle the missing one."""
+    count_last = 0
+    last_frame = velodyne_name
+    while not last_frame in velodyne_names:
+        count_last += 1
+        last_frame = osp.join(
+            velodyne_dir,
+            video_name,
+            f"{str(frame_idx-count_last).zfill(6)}.bin",
+        )
+
+    count_next = 0
+    next_frame = velodyne_name
+    while not next_frame in velodyne_names:
+        count_next += 1
+        next_frame = osp.join(
+            velodyne_dir,
+            video_name,
+            f"{str(frame_idx+count_next).zfill(6)}.bin",
+        )
+
+    if count_last <= count_next:
+        velodyne_name = last_frame
+    else:
+        velodyne_name = next_frame
+
+    return velodyne_name
+
+
+def from_mmpkl_det(
+    data_dir: str,
+    samples_list: str,
+    data_type: str,
+) -> Dataset:
+    """Function converting Glyd MMDetection pkl data to Scalabel format."""
+    frames, groups = [], []
+
+    velodyne_dir = osp.join(data_dir, "velodyne")
+    label_file_pkl = osp.join(data_dir, "pred_instances_3d.pkl")
+    calib_dir = osp.join(data_dir, "calib")
+    ply_dir = osp.join(data_dir, "ply")
+
+    if not osp.exists(ply_dir):
+        os.makedirs(ply_dir)
+
+    labels_list = mmengine.load(label_file_pkl)
+
```

Comment:
Can we add a default value here for convenience?

```suggestion

    if samples_list is None:
        samples_list = osp.join(data_dir, "ImageSets", "train.txt")
        print("Warning: --sample-list not set, defaulting to ", samples_list)